### PR TITLE
NIFI-7932 Allow MonitorActivity's "Threshold Duration" property to use variables

### PR DESCRIPTION
**Pull request by kevinbsilva**
Thank you for submitting a contribution to Apache NiFi.

Please provide a short description of the PR here:

#### Description of PR

When using MonitorActivity, it would be interesting to use variables in "Threshold Duration", which will help maintain Version Control in ProcessGroups (Avoiding "breaking" versions when including the value manually). 

I've included the option expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY), so we can use variables in "Threshold Duration".

Jira Ticket: https://issues.apache.org/jira/browse/NIFI-7932

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [x] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [x] Have you verified that the full build is successful on JDK 8?
- [x] Have you verified that the full build is successful on JDK 11?
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [x] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [x] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [x] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/MonitorActivity.java [hunk 1]
```diff
@@ -85,6 +85,7 @@
             .description("Determines how much time must elapse before considering the flow to be inactive")
             .required(true)
             .addValidator(StandardValidators.TIME_PERIOD_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
```
Comment (kevinbsilva) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/MonitorActivity.java:88]: Added support for Expression Language, within scope VARIABLE_REGISTRY

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/MonitorActivity.java [hunk 2]
```diff
@@ -242,7 +243,7 @@ private boolean shouldReportOnlyOnPrimary(boolean isClusterScope, final ProcessC
 
     @Override
     public void onTrigger(final ProcessContext context, final ProcessSession session) {
-        final long thresholdMillis = context.getProperty(THRESHOLD).asTimePeriod(TimeUnit.MILLISECONDS);
+        final long thresholdMillis = context.getProperty(THRESHOLD).evaluateAttributeExpressions().asTimePeriod(TimeUnit.MILLISECONDS);
```
Comment (kevinbsilva) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/MonitorActivity.java:246]: Added Evaluation for Attribute Expressions (Threshold Duration property)

**Comment by pvillard31** [comment]
Hi @kevinbsilva - thanks for this pull request. The use of parameters is highly recommended over the use of variables going forward. Any reason for not using the parameters in this case? https://nifi.apache.org/docs/nifi-docs/html/user-guide.html#Parameters

**Comment by kevinbsilva** [comment]
Hello @pvillard31, thank you very much for your comment!

There is no particular reason at all, i thought it would be a good idea when using older flows (from imports, etc), especially bigger ones, to avoid problems with version control or even with editing those older flows. 

I didn't know much about parameters, i've read the docs and it helped me a lot. I've shared the knowledge with my team and it was very useful. Thank you very much!

**Comment by markap14** [comment]
I would definitely favor parameters in this situation. In general, we want to avoid having to do lots of code updates to add support for variables for this property and that. But that said, adding support for variables here is also no problem. Since it's already done and it's a trivial change, and all code looks good, will go ahead and merge.

**Pull request merged**